clients/src/main/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegate.java

```diff
@@ -275,4 +282,32 @@ public void onComplete(final ClientResponse response) {
             }
         }
     }
+
+    /**
+     * Creates a {@link Supplier} for deferred creation during invocation by
+     * {@link org.apache.kafka.clients.consumer.internals.DefaultBackgroundThread}.
+     */
+    public static Supplier<NetworkClientDelegate> creator(final Time time,
```

Comment:
Nice integration with the Supplier. Nit: maybe rename this to 'supplier'

Reply:
OK. I didn't like `creator`, so I'm happy to change it 😄